llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -9686,6 +9693,26 @@ SDValue RISCVTargetLowering::lowerINSERT_VECTOR_ELT(SDValue Op,
         return Vec;
       return convertFromScalableVector(VecVT, Vec, DAG, Subtarget);
     }
+
+    // Use ri.vinsert.v.x if available.
+    if (Subtarget.hasVendorXRivosVisni() && VecVT.isInteger() &&
+        isValidInsertExtractIndex(Idx)) {
```

Comment:
If these instructions are meant to be LMUL independent do we want to do the smaller LMUL optimisation? If there's no benefit I'm wondering if this will end up just introducing more vtype toggles

Reply:
This is a good question, one I hadn't considered.  The current scheme is what we've been using downstream and seems to generate decent code, but I'll watch for the vtype issue and may readjust this in a future patch.